### ISSUE TITLE
Model hasDataChanges always true

### ISSUE THREAD

**Issue by Detzler** [magento/magento2#12374]
### Preconditions
1. PHP7.1
2. Magento2.2

### Steps to reproduce

1. Execute:
```
...
$collection = $collectionFactory->create();                              
$collection->addAttributeToSelect('*');
$category = $collection->getFirstItem();
var_dump($category->hasDataChanges());
$category->save();
```

### Expected result

1. hasDataChanges returns false because no data has been changed

### Actual result

1. hasDataChanges returns true
2. Performance impact on $category->save()

**Comment by magento-engcom-team** [comment]
@Detzler, thank you for your report.
We've created internal ticket(s) MAGETWO-84213 to track progress on the issue.

**Comment by virtual97** [comment]
#mageconf

**Comment by magento-engcom-team** [comment]
@virtual97 thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by magento-team** [comment]
Hi @Detzler. Thank you for your report. 
The issue has been fixed in magento/magento2#12736 by @virtual97 in 2.2-develop branch
Related commit(s):
 - a847d89917cca3f1680dbebc2611503a63fa5b1a
 - ae6293108e4d5156a052ebcf4e249ebd06b44d51

The fix will be available with the upcoming patch release.

**Comment by magento-engcom-team** [comment]
Hi @Detzler. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1359 by @magento-engcom-team in 2.3-develop branch
Related commit(s):
 - 1db3f8f6a336026d4a4493c8ba8d3e165ade66d8
 - 412d2287ee5a1a9620fb2a256bab562bdd10fe49
 - 02620b09456d77408ccbbd6a6b61a7b7ebcf73fa
 - 57c5d8a819e403905bd4d5419564d184712bc142
 - 94a3f0346ea086626d8b11c08fd397b11a570caa
 - 0d557dc99c15c6638e1c894ce46cb681a74b8912
 - 48baf590f86590c44bba81a9d5303902aed9d8d7
 - dc3bb78962c4522eb5df8a1a8f40c3bbe2b623fc
 - 0ab45334ef4d44e55c9f84f9c6e4b6a9e08e9237
 - 00f64b077ce00e55216e6ea9fb39071dc74b51a1
 - 440a1b5f9e6309fa87790a5f5b7d1b8fd7ffa53d
 - 4ed8f5b65d5d1779522653102120917c5f9578be
 - 656eeb5587eb1baebaffdc8ba68f1028afda8910
 - fd2f731f7efc4a2cec9e03f8de88bf39c0988829
 - 2c1a23faa60805bd63ff62216309d57a052612d1
 - f7ff71a92bcf56679b82c22549244e1adb9f4ed2
 - ac9bcfa15b1b0b98f9cb3851f4d9b56e18ede89b
 - 55debe0c2b514e37271ed6edf20b95fec6cc41c9
 - 349508628dc1c4334e13b3541227505e26c93c64
 - d678f1568b6e2d30c6d651dd3da02e0f3512353e
 - bf8ee3089f005e267f0b14cf81b14e9a3ee26986
 - 335c91c07dde6f90309f11cf0efcbd9127630955
 - c5a823f95161f7c13fb0947abb445fdc9e9a620a
 - 696a5a575513bb87c47ffc15a571034b0b91d7ae

The fix will be available with the upcoming 2.3.0 release.

**Comment by mhafizumer** [comment]
Still issue, `$product->hasDataChanges()` return me true whenever I edit or not edit any attribute data.
I am using magento 2.4.1 CE
